lib/reporter/index.js

```diff
@@ -195,10 +195,15 @@ class Reporter extends SimplifiedReporter {
           result.status === -1
         );
 
-      // Use only necessary values 
+      // Use only necessary values
       if (result) {
+        if (result.constructor.name === 'NightwatchAPI') {
```

Comment:
I am having trouble wrapping my head around this.. @garg3133 can you explain briefly how this lets the reporter wait for the test to finish?

Reply:
It does not do that. And if I'm not wrong, the global reporter does not _wait_ for every test to finish, it just logs whatever reports it gets during the run-time of the main process.

So, what was happening here before was that when the worker was trying to send the report to the main process, it was failing if there was even one function anywhere in the report object, and hence no report was reaching the main process.

The above change merely helps to avoid the generation of that function in the report object due to which it was failing. I have not taken the positive approach here (allow only a few results to pass through, e.g. `result.construction.name === 'Object`, because there can be many valid result objects (`Object` instance, `Error` instance, `NightwatchAssertionFailure` instance, etc. but only a few (right now only one) which results in error).

Reply:
Got it, thanks for the explanation.

Reply:
After the recent changes, this is not required anymore and can be removed. I've only kept it because if the result is `NightwatchAPI`, we ideally should not destruct it.

Reply:
We can remove this.

Reply:
Done